### PR TITLE
Add transaction options to Keycloak CR

### LINKED ISSUE #14375: Add transaction options to Keycloak CR
Mapped options:
* transaction-xa-enabled

Example CR:
```yaml
apiVersion: k8s.keycloak.org/v2alpha1
kind: Keycloak
metadata:
  name: example-kc
spec:
  transaction:
    xa-enabled: false # boolean
...
```

### PR DESCRIPTION
**Pull request by mabartos**
Closes #14375

### CODE REVIEW

File: operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/TransactionsSpec.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import io.sundr.builder.annotations.Buildable;
+
+import java.io.Serializable;
+
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class TransactionsSpec implements Serializable {
+
+    @JsonProperty("xa-enabled")
+    @JsonPropertyDescription("Determine whether Keycloak should use a non-XA datasource in case the database does not support XA transactions.")
+    private Boolean xaEnabled;
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/TransactionsSpec.java:31]: I believe we should not be using kebab case here (even though it's native for Quarkus dist). This is an K8s-native API so we should use camel case. (It was wrongly stated in the [original issue](https://github.com/keycloak/keycloak/issues/14375) to use `xa-enabled`.)

Thank you!

Comment (mabartos) [reply operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/TransactionsSpec.java:31]: You're right! Thanks!

**Review by vmuzikar — APPROVED** [review]
LGTM, thank you @mabartos!

Let's wait for the CI to finish...

**Pull request merged**